Add transformation to tortoise radius

nilsvu opened the pull request:
## Proposed changes

Bundles some root-finding code from https://github.com/tosburn3/KerrGeodesicsC to transform between Boyer-Lindquist and tortoise radius. The bundled code has figured out good initial guesses and a robust method for the root find.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Add transformation to tortoise radius

wthrowe commented:
I think this is a license violation.  The GPL would let us distribute a verbatim copy of the upstream package, but I don't think we can distribute a version that has been modified like this as part of SpECTRE.

nilsvu commented:
My understanding is that this is a redistribution of the GPL-licensed code with some changes still under the GPL license, which is ok. The changes are: Deleted some functions to avoid dependence on FFTW, added a CMakeLists.txt to compile the code with CMake, added a C++ compatibility header, and auto-formatted the code. These are listed in the README, and the license is also included.

wthrowe commented:
The GPL does not give you permission to do that.  You must license the entire work as GPL if you include modified GPL code, not just your changes.  Since we're not relicensing SpECTRE, your options are:
1. Use the unmodified GPL code, so the modification clauses do not come into effect,
2. Convince the author to relicense, or
3. Use TOMS748 instead of including an external package just to do a 1d root find.

nilsvu committed:
Improve TOMS748 errors

wthrowe reviewed (approved):
Looks good.

Noting in case someone later wants to do bounds optimization: For large radius, good bounds look to be `0 < r - (r* - 2 M log(r*/(2 M))) < const ~ 1.5 M`.  I didn't experiment with small radius.
